testutils/sqlutils/conn.go

```diff
@@ -47,9 +48,9 @@ func PGUrl(t util.Tester, ts *server.TestServer, user, tempDir, prefix string) (
 
 	// Copy these assets to disk from embedded strings, so this test can
 	// run from a standalone binary.
-	tempCAPath, tempCACleanup := securitytest.RestrictedCopy(t, caPath, tempDir, "TestLogic_ca")
-	tempCertPath, tempCertCleanup := securitytest.RestrictedCopy(t, certPath, tempDir, "TestLogic_cert")
-	tempKeyPath, tempKeyCleanup := securitytest.RestrictedCopy(t, keyPath, tempDir, "TestLogic_key")
+	tempCAPath, tempCACleanup := securitytest.RestrictedCopy(t, caPath, tempDir, strings.Join([]string{user, prefix, "ca"}, "_"))
```

Comment:
These filenames are still not unique, are they? This greatly reduces the chance of a race since different tests won't affect each other, but multiple processes running the same test (as in a `stress` run) still could.